hip-362-token-freeze-amount.md

```diff
@@ -0,0 +1,43 @@
+**hip: TBD </br>
+title: Integrate '.setTokenAmount()' custom function for 'TokenFreezeTransaction()' </br>
+author: Tomachi Anura [shibartoken@protonmail.com](mailto:shibartoken@protonmail.com)</br>
+type: Standards Track</br>
+category: Service</br>
+needs-council-approval: Yes</br>
+status: Draft</br>
+last-call-date-time:</br>
+created: 22.02.13</br>
+discussions-to:**</br>
+
+## Abstract
+
+When "locking" or "freezing" liquidity related to an associated Account Id, There is no way to freeze a specific amount of tokens. The Shibar Network Team strongly believes that it would bring many advantages to have the tokenFreezeTransaction like in the sequent example:
+
+<img width="711" alt="Screen Shot 2022-02-13 at 5 44 48 PM" src="https://user-images.githubusercontent.com/96840872/153778591-6ba9b8b7-92a7-4438-b9fb-23e77a469e4d.png">
+
+## Motivation
+
+When "locking" or "freezing" liquidity related to an associated Account Id, There is no way to freeze a specific amount of tokens. This could be potentially detrimental in regards of DeFi, if the same token offers many use-cases. As it stands now, Users would have to set different accounts in order to use the same Network.
+
+
+## Rationale
+When Alice participates to a round 1 token pre-sale, she buys 1000 tokens. According to the terms and conditions of the round 1 pre-sale, she should have those tokens frozen and unable to move them from the wallet until the term expires. However Alice would also like to buy more tokens at round 2 which are supposed to be Unfrozen tokens. Unfortunately Hedera only allows to freeze the whole token amount of the account, while it would be much more beneficial to implement a child function .setTokenAmount(Amount) to the parent TokenFreezeTransaction().
+
+This implementation allows Alice to interact with the token if she holds more than the frozen amount. for many use cases, without limiting her whole account by keeping the whole token amount frozen. This way, she can buy another 1000 tokens on round 2, having a total balance of 2000. And she is able to use unfrozen 1000 tokens from round 2 to interact with the network
+
+## User Stories
+
+- As dApp, I would like to allow users to stake their tokens by freezing a partial amount of TokenX in the user wallet for a pre-determined duration
+## Specification
```

Comment:
Please add the Specification section information to progress with the HIP!

Reply:
We need more clarification here in this wrt the motivation section to get a better understanding for all who come to read this User story specifically. In the motivation we talk about freezing tokens in an account because of several presale rounds but then in this User story we mention them being staked via a TokenFreeze txn. I feel like a little more clarification would go a long way for people like me who come to check this out.

Reply:
as a token owner, i wish to have the chance to freeze a specific amount of tokens, instead to freeze/unfreeze them all at once like it acts right now.

Reasons:
- you can buy 1000 token in a presale, and the condition is that they are locked for 1 year. 
you can then buy 2000 token in the public sale, and those tokens won't be locked at all.
Right now, we are forced to ask our buyers to setup 2 wallets, so we lock the 1000tokens in the first wallet, and we let the 2000 tokens unlocked in the second one.
With this HIP, we might be facilitating the customer experience, and also we can better handle from our end the lock amount.

Possible uses:
- private/public sales with different locking periods
- changes to implement a NON custodial staking mechanism
- changes to slowly release the locked amount with programmatic cron jobs
so let's say you want to stake the 50% of you Hsuite, you don't even have to send them to use (we have no smart contracts, as you know)